src/client/sound/s_sample.c

```diff
@@ -28,47 +28,109 @@ static const char *SAMPLE_TYPES[] = { ".ogg", ".wav", NULL };
 static const char *SOUND_PATHS[] = { "sounds/", "sound/", NULL };
 
 /**
+ * @brief Resample audio.
+ */
+size_t S_Resample(const int32_t inrate, const int32_t outrate, const size_t incount, const int16_t *indata, int16_t **outdata) {
+	const vec_t stepscale = (vec_t) inrate / (vec_t) outrate;
+	const size_t outcount = incount / stepscale;
+
+	*outdata = Mem_Malloc(outcount * sizeof(int16_t));
+
+	int32_t samplefrac = 0;
+	const vec_t fracstep = stepscale * 256.0;
+
+	for (size_t i = 0; i < outcount; i++) {
+		const int32_t srcsample = samplefrac >> 8;
+
+		samplefrac += fracstep;
+		(*outdata)[i] = LittleShort(indata[srcsample]);
+	}
+
+	return outcount;
+}
+
+/**
+ * @brief Convert from stereo to mono.
+ */
+/*static size_t S_Monoize(const size_t incount, const int16_t *indata, int16_t **outdata) {
+	const size_t outcount = incount / 2;
+
+	*outdata = Mem_Malloc(outcount * sizeof(int16_t));
+
+	for (size_t i = 0, x = 0; x < outcount; i += 2, x++) {
+		(*outdata)[x] = (((int32_t) indata[i]) + ((int32_t) indata[i + 1])) / 2;
+	}
+
+	return outcount;
+}*/
```

Comment:
Ideally it'd only be called on channels == 2, and I don't think we'd ever support > 2